build-tools/xabootstrap/xabootstrap/ConfigAndData/Dependencies/AndroidToolchain.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Xamarin.Android.Bootstrap
+{
+	partial class AndroidToolchain : AppObject
+	{
+		public static readonly Uri AndroidUri = Configurables.Urls.AndroidToolchain_AndroidUri;
+		public static readonly Uri AntUri     = Configurables.Urls.AndroidToolchain_AntUri;
+
+		public List<AndroidToolchainComponent> Components { get; }
+
+		public AndroidToolchain ()
+		{
+			string AndroidNdkVersion       = BuildAndroidPlatforms.AndroidNdkVersion;
+			string AndroidNdkDirectory     = GetRequiredProperty (KnownProperties.AndroidNdkDirectory);
+			string AndroidCmakeVersion     = GetRequiredProperty (KnownProperties.AndroidCmakeVersion);
+			string AndroidCmakeVersionPath = GetRequiredProperty (KnownProperties.AndroidCmakeVersionPath);
+			string EmulatorVersion         = GetRequiredProperty (KnownProperties.EmulatorVersion);
+			string XABuildToolsFolder      = GetRequiredProperty (KnownProperties.XABuildToolsFolder);
+			string XABuildToolsVersion     = GetRequiredProperty (KnownProperties.XABuildToolsVersion);
+			string XAPlatformToolsVersion  = GetRequiredProperty (KnownProperties.XAPlatformToolsVersion);
+
+			Components = new List<AndroidToolchainComponent> {
+				new AndroidPlatformComponent ("android-2.3.3_r02-linux", "10"),
+				new AndroidPlatformComponent ("android-15_r03",          "15"),
+				new AndroidPlatformComponent ("android-16_r04",          "16"),
+				new AndroidPlatformComponent ("android-17_r02",          "17"),
+				new AndroidPlatformComponent ("android-18_r02",          "18"),
+				new AndroidPlatformComponent ("android-19_r03",          "19"),
+				new AndroidPlatformComponent ("android-20_r02",          "20"),
+				new AndroidPlatformComponent ("android-21_r02",          "21"),
+				new AndroidPlatformComponent ("android-22_r02",          "22"),
+				new AndroidPlatformComponent ("platform-23_r03",         "23"),
+				new AndroidPlatformComponent ("platform-24_r02",         "24"),
+				new AndroidPlatformComponent ("platform-25_r03",         "25"),
+				new AndroidPlatformComponent ("platform-26_r02",         "26"),
+				new AndroidPlatformComponent ("platform-27_r03",         "27"),
+				new AndroidPlatformComponent ("platform-28_r04",         "28"),
+
+				// arguments are: string name, string destDir, Uri relativeUrl = null, bool isMultiVersion = false, bool noSubdirectory = false
+				{ "docs-24_r01", "docs"},
+				{ "android_m2repository_r16", Path.Combine ("extras", "android", "m2repository")},
+				{ "x86-28_r04", Path.Combine ("system-images", "android-28", "default", "x86"), new Uri ("sys-img/android/", UriKind.Relative)},
+				{$"android-ndk-r{AndroidNdkVersion}-{osTag}-x86_64", AndroidNdkDirectory},
```

Comment:
Apparently `AndroidNdkVersion` isn't set when expected/required, because when [the build attempts to download this url](https://jenkins.mono-project.com/job/xamarin-android-pr-pipeline-release/837/console), it fails:

```
error : Unable to download URL `https://dl.google.com/android/repository/android-ndk-r-darwin-x86_64.zip` to `/Users/builder/android-archives/android-ndk-r-darwin-x86_64.zip`: Response status code does not indicate success: 404
```

Reply:
`BuildAndroidPlatforms.AndroidNdkVersion ` is `const`; how could this *not* be set/available here?

Reply:
The build fails because it uses `make prepare` and not the new bootstrapper which changed the way NDK references work. At this moment the build failure is expected, until I make the new bootstrapper the "default" in this PR